ReactiveCocoa/Swift/Signal.swift

```diff
@@ -569,6 +569,50 @@ public func takeWhile<T, E>(predicate: T -> Bool)(signal: Signal<T, E>) -> Signa
 	}
 }
 
+/// Throttle values sent by the receiver, so that at least `interval`
+/// seconds pass between each, then forwards them on the given scheduler.
+///
+/// If multiple values are received before the interval has elapsed, the
+/// latest value is the one that will be passed on.
+///
+/// If the input signal completes while a value is being throttled, that value
+/// will be discarded and the returned signal will complete immediately.
+public func throttle<T, E>(interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
+
+	return Signal { observer in
+		let previousDate: Atomic<NSDate?> = Atomic(nil)
+		let schedulerDisposable = SerialDisposable()
+
+		let disposable = CompositeDisposable()
+		disposable.addDisposable(schedulerDisposable)
+
+		let signalDisposable = signal.observe(next: { value in
+			let scheduleDate = previousDate.value?.dateByAddingTimeInterval(interval) ?? scheduler.currentDate
+
+			schedulerDisposable.innerDisposable = scheduler.scheduleAfter(scheduleDate) {
+				previousDate.value = scheduler.currentDate
+
+				// There's a small race where a new disposable could be set
+				// before we've updated the `previousDate`, which would schedule
+				// a value too early. Disallow that.
+				schedulerDisposable.innerDisposable = nil
```

Comment:
I was more than a bit distracted when I wrote this, and I'm realizing now that it could result in values being lost if they're sent in that race condition window. I'll fix this up.